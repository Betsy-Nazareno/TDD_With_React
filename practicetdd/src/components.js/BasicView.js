import React, {Component} from 'react';

class BasicView extends Component{

    constructor(props){
        super(props);
        this.onClick = this.onClick.bind(this);
        const {disabled=false} = this.props;
        this.state = {
            value: 1,
            disabled: disabled
        }
    }

    onClick(){
        this.props.callAPI(1, 2, 3);
    }

    render(){
        const { b:c = {} } = this.props;
        const { a } = c;
        const {disabled} = this.state;

        return(
            <div>
                Hello world {a}
                <span onClick={this.onClick}>
                    hola
                </span>
                <span>
                    {disabled? 'ENABLE': 'DISABLE'}
                </span>
            </div>
    
        )

    }

}


export default BasicView;