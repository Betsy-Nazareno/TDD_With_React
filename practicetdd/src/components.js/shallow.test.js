import React from 'react';
import {shallow} from 'enzyme';
import BasicView from './BasicView';
import toJson from 'enzyme-to-json';

describe('Shallow tests', ()=>{
    test('test 1', ()=>{
        const wrapper = shallow(<BasicView/>)
        expect(wrapper.find('div').length).toEqual(1);

    });

    test('test 2', ()=>{
        const wrapper = shallow(<BasicView/>)

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('test 3', ()=>{

        const callAPI = jest.fn();
        const wrapper = shallow(<BasicView callAPI={callAPI} b={{a:"betsy"}}/>)
        expect(callAPI).toHaveBeenCalledTimes(0);

        wrapper.find('span').first().simulate('click');
        expect(callAPI).toHaveBeenCalledTimes(1);
        expect(callAPI).toBeCalledWith(1, 2, 3);

    });

    test('test 4', ()=>{
        const wrapper = shallow(<BasicView b={{a:"betsy"}}/>)
        expect(wrapper.instance().state).toEqual({value: 1, disabled: false})

    });
    test('test 5', ()=>{
        const wrapper = shallow(<BasicView disabled={false} b={{a:"betsy"}}/>)
        expect(wrapper.instance().state).toEqual({value: 1, disabled: false})

    });
    test('test 6', ()=>{
        const wrapper = shallow(<BasicView disabled={false} b={{a:"betsy"}}/>)

        const span = wrapper.find('span').at(1);
        expect(span.text()).toEqual('DISABLE'); 
    });
    test('test 7', ()=>{
        const wrapper = shallow(<BasicView disabled={false} b={{a:"betsy"}}/>)

        wrapper.instance().setState({disabled: true})
        const span = wrapper.find('span').at(1);
        expect(span.text()).toEqual('ENABLE'); 
    });
})

