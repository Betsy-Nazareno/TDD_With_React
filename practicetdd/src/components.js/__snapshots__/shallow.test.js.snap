// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow tests test 2 1`] = `
<div>
  Hello world 
  <span
    onClick={[Function]}
  >
    hola
  </span>
  <span>
    DISABLE
  </span>
</div>
`;
